src/main/java/HandleGetRequest.java

```diff
@@ -0,0 +1,74 @@
+package com.td.HttpServer;
+
+import java.util.*;
+import java.io.IOException;
+
+public class HandleGetRequest extends HttpHandleRequest {
+
+  private IFileIO fileIO;
+
+  public HandleGetRequest(IFileIO fileIO) {
+    super();
+    this.fileIO = fileIO;
+  }
+
+  public HttpResponse generateResponse(HttpRequest request) {
+    String path = request.path();
+    if (path.equals("/")) {
+      return defaultResponse();
+    }
+    try {
+      byte[] body = fileIO.getContent(path);
+      HashMap<String, String> headers = buildHeaders(body, path);
+      return new HttpResponse(CODE_200, body, headers);
+    }
+    catch (IOException e) {
+      return notFound(request);
+    }
+  }
+
+  private HttpResponse defaultResponse() {
+    byte[] body = "Default path / ".getBytes();
+    HashMap<String, String> headers = buildHeaders(body, "/");
+    return new HttpResponse(CODE_200, body, headers);
+  }
+
+  private HttpResponse notFound(HttpRequest request) {
+    byte[] body = "File not Found".getBytes();
+    HashMap<String, String> headers = buildHeaders(body, request.path());
+    return new HttpResponse(CODE_404, body, headers);
+  }
+
+  private HashMap<String, String> buildHeaders(byte[] body, String path) {
+    HashMap<String, String> rtnMap = new HashMap<String, String>();
+    rtnMap.putAll(standardHeaders());
+    rtnMap.putAll(contentHeaders(body, path));
+    return rtnMap;
+  }
+
+  private HashMap<String, String> contentHeaders(byte[] body, String path) {
+    HashMap<String, String> rtnMap = new HashMap<String, String>();
+    rtnMap.put("Content-Type", contentType(path));
+    rtnMap.put("Content-Length", contentLength(body));
+    return rtnMap;
+  }
+
+ private String contentLength(byte[] body) {
+    return Integer.toString(body.length);
+  }
+
+  private String contentType(String fileName) {
```

Comment:
Gotcha. I could probably make a HashMap<String, String> with file extensions as key and the content type as the value. That's the first thing that comes to mind to me as to how to clean this up. Would that be a good idea, or would you suggest something else that I might consider?